jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/java/codeaction/InsertAnnotationAttributeQuickFix.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4jakarta.jdt.core.java.codeaction;
+
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Annotation;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4jakarta.commons.codeaction.CodeActionResolveData;
+import org.eclipse.lsp4jakarta.commons.codeaction.ICodeActionId;
+import org.eclipse.lsp4jakarta.jdt.core.java.corrections.proposal.ChangeCorrectionProposal;
+import org.eclipse.lsp4jakarta.jdt.core.java.corrections.proposal.InsertAnnotationAttributeProposal;
+
+/**
+ * QuickFix for inserting attribute of a given annotation.
+ *
+ * @author Angelo ZERR
+ *
+ */
+public abstract class InsertAnnotationAttributeQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(InsertAnnotationAttributeQuickFix.class.getName());
+
+    private static final String CODE_ACTION_LABEL = "Insert ''{0}'' attribute";
```

Comment:
Let me address all of your concerns here because they are related.
The code actions under `*/jdt/core/java/codeaction` are generic in nature. They are and could be extended by specific implementations. They might, in theory, belong in a common quick fix utils type folder in a common repository.
Classes such as InsertAnnotationAttributeQuickFix and InsertAnnotationMissingQuickFix are files that were ported and are generic preexisting classes that define the quick fix name/label in the class itself. If these and the newly added generic code action are moved to a common project, they would need to have their own messages.properties resource file.
The reason you see many messages removed is because they are using (extending) these generic quick fix implementations and labels. Some of the original messages have changed slightly because of that.